File: AFMobile/AFAndroid/app/src/main/java/cz/cvut/fel/matyapav/afandroid/builders/widgets/types/BasicBuilder.java
package cz.cvut.fel.matyapav.afandroid.builders.widgets.types;

import android.app.Activity;
import android.view.View;

import cz.cvut.fel.matyapav.afandroid.components.parts.AFField;
import cz.cvut.fel.matyapav.afandroid.components.parts.FieldInfo;
import cz.cvut.fel.matyapav.afandroid.components.skins.Skin;

/**
 * Created by Pavel on 24.02.2016.
 */
abstract class BasicBuilder implements AbstractBuilder {

    private Skin skin;

    private FieldInfo properties;

    public BasicBuilder(Skin skin, FieldInfo properties) {
        this.skin = skin;
        this.properties = properties;
    }

    public Skin getSkin() {
        return skin;
    }

    public FieldInfo getProperties() {
        return properties;
    }
}

File: AFMobile/AFWindowsPhone/AFWindowsPhone/builders/widgets/BasicBuilder.cs
using AFWindowsPhone.builders.components.parts;
using AFWindowsPhone.builders.skins;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Windows.UI.Xaml;

namespace AFWindowsPhone.builders.widgets
{
    abstract class BasicBuilder : AbstractWidgetBuilder
    {
        private Skin skin;
        private FieldInfo properties;

        public BasicBuilder(Skin skin, FieldInfo properties)
        {
            this.skin = skin;
            this.properties = properties;
        }

        public Skin getSkin()
        {
            return skin;
        }

        public FieldInfo getProperties()
        {
            return properties;
        }

        public abstract FrameworkElement buildFieldView();
        public abstract void setData(AFField field, object value);
        public abstract object getData(AFField field);
    }
}
